src/query/scan_manager.c

```diff
@@ -7637,3 +7734,317 @@ scan_print_stats_text (FILE * fp, SCAN_ID * scan_id)
+      return S_SUCCESS;
+    }
+
+  return qp_scan;
+}
+
+/*
+ * scan_next_hash_list_scan () - The scan is moved to the next hash list scan item.
+ *   return: SCAN_CODE (S_SUCCESS, S_END, S_ERROR)
+ *   scan_id(in/out): Scan identifier
+ *
+ * Note: If there are no more scan items, S_END is returned. If an error occurs, S_ERROR is returned.
+ */
+static SCAN_CODE
+scan_hash_probe_next (THREAD_ENTRY * thread_p, SCAN_ID * scan_id, QFILE_TUPLE * tuple)
+{
+  LLIST_SCAN_ID *llsidp;
+  SCAN_CODE qp_scan;
+  HASH_SCAN_KEY *key;
+  HASH_SCAN_VALUE *hvalue;
+  QFILE_LIST_SCAN_ID *scan_id_p;
+
+  llsidp = &scan_id->s.llsid;
+  key = llsidp->hlsid.temp_key;
+  scan_id_p = &llsidp->lsid;
+
+  if (scan_id_p->position == S_BEFORE)
+    {
+      if (llsidp->hlsid.hash_table->nentries > 0)
+	{
+	  /* init curr_hash_entry */
+	  llsidp->hlsid.curr_hash_entry = NULL;
+	  /* build key */
+	  if (qdata_build_hscan_key (thread_p, scan_id->vd, llsidp->hlsid.probe_regu_list, key) != NO_ERROR)
+	    {
+	      return S_ERROR;
+	    }
+
+	  /* get value from hash table */
+	  hvalue =
+	    (HASH_SCAN_VALUE *) mht_get2 (llsidp->hlsid.hash_table, key, (void **) &llsidp->hlsid.curr_hash_entry);
+	  if (hvalue == NULL)
+	    {
+	      return S_END;
+	    }
+	  *tuple = hvalue->tuple;
+	  scan_id_p->position = S_ON;
+	  return S_SUCCESS;
+	}
+      else
+	{
+	  return S_END;
+	}
+    }
+  else if (scan_id_p->position == S_ON)
+    {
+      if (llsidp->hlsid.curr_hash_entry->next)
+	{
+	  llsidp->hlsid.curr_hash_entry = llsidp->hlsid.curr_hash_entry->next;
+	  *tuple = ((HASH_SCAN_VALUE *) llsidp->hlsid.curr_hash_entry->data)->tuple;
+	  return S_SUCCESS;
+	}
+      else
+	{
+	  scan_id_p->position = S_AFTER;
+	  return S_END;
+	}
+    }
+  else if (scan_id_p->position == S_AFTER)
+    {
+      return S_END;
+    }
+  else
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_QPROC_UNKNOWN_CRSPOS, 0);
+      return S_ERROR;
+    }
+
+  return qp_scan;
+}
+
+/*
+ * check_hash_list_scan () - Check if hash list scan is possible
+ *   return: bool
+ *   llsidp (in): list scan id pointer
+ *   node :
+ *      1. count of tuple of list file > 0
+ *      2. list file size check
+ *      3. regu_list_build, regu_list_probe is not null
+ *      4. The number of probe regu_var and build regu match
+ *      5. type of regu var is not oid && vobj
+ *      6. list file from dptr is not allowed
+*/
+static bool
+check_hash_list_scan (LLIST_SCAN_ID * llsidp, int *val_cnt, int hash_list_scan_yn)
+{
+  int build_cnt;
+  regu_variable_list_node *build, *probe;
+  DB_TYPE vtype1, vtype2;
+  UINT64 mem_limit = prm_get_bigint_value (PRM_ID_MAX_HASH_LIST_SCAN_SIZE);
+
+  /* no_hash_list_scan sql hint check */
+  if (hash_list_scan_yn == 0)
+    {
+      return false;
+    }
+
+  /* count of tuple of list file > 0 */
+  if (llsidp->list_id->tuple_cnt <= 0)
+    {
+      return false;
+    }
+  /* list file size check */
+  if ((UINT64) (llsidp->list_id->page_cnt * DB_PAGESIZE) > mem_limit)
```

Comment:
`llsidp->list_id->page_cnt * DB_PAGESIZE` may occur overflow and it is just casted to UINT64.

Reply:
I fixed it.